src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -118,12 +189,205 @@
+
+      // Find the contest for tabulation
+      Contest contestToTabulate = null;
+      for (Election election : cvrReport.Election) {
+        for (Contest contest : election.Contest) {
+          if (contest.Name.equals(this.contestId)) {
+            contestToTabulate = contest;
+            break;
+          }
+        }
+      }
+
+      if (contestToTabulate == null) {
+        Logger.severe("Contest \"%s\" from config file not found!", this.contestId);
+        throw new CvrParseException();
+      }
+
+      // build a map of Candidates
+      HashMap<String, Candidate> candidateById = new HashMap<>();
+      for (Election election : cvrReport.Election) {
+        for (Candidate candidate : election.Candidate) {
+          candidateById.put(candidate.ObjectId, candidate);
+        }
+      }
+
+      // build a map of ContestSelections
+      HashMap<String, ContestSelection> contestSelectionById = new HashMap<>();
+      for (ContestSelection contestSelection : contestToTabulate.ContestSelection) {
+        if (contestSelection.CandidateIds.length > 1) {
+          Logger.warning(
+              "Multiple CandidateIds found in ContestSelection \"%s\". "
+                  + " Only the first one will be processed.",
+              contestSelection.ObjectId);
+        }
+        contestSelectionById.put(contestSelection.ObjectId, contestSelection);
+      }
+
+      // build a map of GpUnits (aka precinct or district)
+      HashMap<String, GpUnit> gpUnitById = new HashMap<>();
+      for (GpUnit gpUnit : cvrReport.GpUnit) {
+        gpUnitById.put(gpUnit.ObjectId, gpUnit);
+      }
+
+      // process the Cvrs
+      for (CVR cvr : cvrReport.CVR) {
+        CVRContest contest = getCvrContest(cvr, contestToTabulate);
+        if (contest == null) {
+          // the CVR does not contain any votes for this contest
+          continue;
+        }
+        List<Pair<Integer, String>> rankings = new ArrayList<>();
+        // parse CVRContestSelections into rankings
+        // they will be null for an undervote
+        if (contest.CVRContestSelection != null) {
+          for (CVRContestSelection cvrContestSelection : contest.CVRContestSelection) {
+            if (cvrContestSelection.Status != null && cvrContestSelection.Status
+                .equals(STATUS_NEEDS_ADJUDICATION)) {
+              Logger.info("Contest Selection needs adjudication.  Skipping.");
+              continue;
+            }
+            String contestSelectionId = cvrContestSelection.ContestSelectionId;
+            ContestSelection contestSelection = contestSelectionById
+                .get(contestSelectionId);
+            if (contestSelection == null) {
+              Logger.severe("ContestSelection \"%s\" from CVR not found!", contestSelectionId);
+              throw new CvrParseException();
+            }
+            String candidateId;
+            // check for declared write-in:
```

Comment:
You say "declared" here, but then you use the undeclared write-in label. Just a typo in your comment?